src/init/starship.xsh

```diff
@@ -0,0 +1,15 @@
+import random
+import string
+
+
+def starship_prompt():
+    last_cmd = __xonsh__.history[-1] if __xonsh__.history else None
+    status = last_cmd.rtn if last_cmd else 0
+    jobs = len(__xonsh__.all_jobs)
```

Comment:
After running any command the jobs will sometimes display for me, with only the starship init in my `.xonshrc`.

Reply:
[Fixed](https://github.com/starship/starship/pull/2807/commits/57ea974f84017c53f5bf8fc33f27894205180003).

Getting job accounting right in xonsh apparently requires regularly calling [`_clear_dead_jobs`](https://github.com/xonsh/xonsh/blob/c06ed5ef2dcc44ba7a4b1c3522f0864cddffef81/xonsh/jobs.py#L265-L273).

We could accomplish that here by calling [`get_next_job_number`](https://github.com/xonsh/xonsh/blob/c06ed5ef2dcc44ba7a4b1c3522f0864cddffef81/xonsh/jobs.py#L291-L297). In fact, #1265 did [exactly that](https://github.com/starship/starship/blob/f34be39da2230cde361f8c8611e95d95b52b8326/src/init/starship.xsh#L7).

Unfortunately, due to what I [believe](https://gitter.im/xonsh/xonsh?at=60e8832d82dd9050f5e0c96a) is the [same](https://github.com/starship/starship/pull/1265#pullrequestreview-420227964 ) issue as [above](https://github.com/starship/starship/pull/2807#discussion_r666360733), we're pretty limited here: we can't use any imported functions or even any names we define outside of this function. As a workaround, I've duplicated enough of the relevant logic in `_clear_dead_jobs` to ignore dead jobs in our count here.

> After running any command the jobs will sometimes display for me, with only the starship init in my .xonshrc.

Incidentally, I reproduced this on xonsh 0.9.27 but not on xonsh's latest main branch. But I don't understand why they would be different. 🤷🏻‍♂️